tests/unittest_inference.py

```diff
@@ -3808,6 +3808,50 @@ class A:  pass
         with pytest.raises(InferenceError):
             next(ast_node5.infer())
 
+        ast_nodes6 = extract_node(
```

Comment:
This test doesn't fail on `main` for me. 🤔

Reply:
Yeah, I was just writing some tests for `__new__` and though it would be good to keep this in as a some sort of regression tests.

Reply:
I've struggling to do so. I'll give it another try. I can probably hack something together with `getattr("__new__")` on `super()`.

Reply:
No leads from the primer examples (too complicated)?

Reply:
Not really. Those fails are dependent on what `pylint` does with the information it gets from `astroid`. Using `getattr` we can test that information here.